src/config.rs

```diff
@@ -367,6 +375,126 @@ pub(crate) struct GooseDefaults {
     pub manager_port: Option<u16>,
 }
 
+/// Internal data structure representing a test plan.
+//#[derive(Options, Debug, Clone, Serialize, Deserialize)]
+#[derive(Options, Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct TestPlan {
+    // A test plan is a vector of tuples each indicating a # of users and milliseconds.
+    pub(crate) steps: Vec<(usize, usize)>,
+    // Which step of the test_plan is currently running.
+    pub(crate) current: usize,
+}
+
```

Comment:
I would put test plan and its logic in a separate file. I would also include `TestPlanStepAction` and `TestPlanHistory` that are currently in `metrics.rs`.

Reply:
Besides that I'd also add some test coverage.

Reply:
There's very little logic here, but I suppose it's likely going to grow -- splitting it out works for me. (Especially as when I was writing this I could never remember which file it lived in).

And yes, it DEFINITELY needs tests! :)

Reply:
I've split it into its own file now -- definitely the right call, thanks.